### PR TITLE
WithinIntervalCensororing

### PR DESCRIPTION
**Pull request by medewitt**
## Description

Work to make the WithinIntervalCensoring for some values. Implements:

* A struct called WithinIntervalCensoredDist
* Constructor helper
* Method to generate pmf

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: src/WithinIntervalCensored.jl [hunk 1]
```diff
@@ -0,0 +1,76 @@
+@doc raw"
+Implement a censoring function. 
+Takes as and input the the object from truncated form of the
```
Comment (seabbs) [inline src/WithinIntervalCensored.jl:3]:
```suggestion
Takes a `UnivariateDistribution` as an input and depends on that distributions cdf method.
```

File: src/WithinIntervalCensored.jl [hunk 2]
```diff
@@ -0,0 +1,76 @@
+@doc raw"
+Implement a censoring function. 
+Takes as and input the the object from truncated form of the 
+primarycensored object. This object only needs to have a method
```
Comment (seabbs) [inline src/WithinIntervalCensored.jl:4]:
```suggestion
```

File: src/WithinIntervalCensored.jl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+for the cumulative distribution function.
```
Comment (seabbs) [inline src/WithinIntervalCensored.jl:5]:
```suggestion
```

File: src/WithinIntervalCensored.jl [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+- `dist` the truncated primarycensored distribution
+- `obs_time` the observation time of interest
+- `swindow` the censoring window for the interval
+- `pwindow` primary censoring window
+
+# Returns
+- WithinIntervalCensoredDist object
+
+"
+#! TODO Struct
```
Comment (seabbs) [inline src/WithinIntervalCensored.jl:17]:
```suggestion
```

**Review by medewitt — COMMENTED** [review]
looks god

**Pull request merged**